docs/exts/xref.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+from docutils import nodes
+
+from sphinx.util import caption_ref_re
+
+
+def xref(typ, rawtext, text, lineno, inliner, options={}, content=[]):
+
+    title = target = text
+    titleistarget = True
+    # look if explicit title and target are given with `foo <bar>` syntax
+    brace = text.find('<')
+    if brace != -1:
+        titleistarget = False
+        m = caption_ref_re.match(text)
```

Comment:
Nitpick: Unless variables are inside a list comprehension or some generator expression I'd ask that we abandon single letter variables. (This affects the following 3 lines too).

```suggestion
        match = caption_ref_re.match(text)
```

Reply:
We can't change this code. https://github.com/michaeljones/sphinx-xref/blob/master/xref.py

Reply:
Okay. Since it isn't clear from the PR code can you add a short paragraph to the PR body about what `xref` is doing?